### PR TITLE
Different way to do redactions on meta fields

### PR DESCRIPTION
**Pull request by benjdlambert**
- feat: fix issue with redaction by implementing another way
- chore: safer way to do redactions

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend-app-api/src/logging/WinstonLogger.ts [hunk 1]
```diff
@@ -85,24 +86,17 @@ export class WinstonLogger implements RootLoggerService {
...
-    };
     return {
-      format: format(replace)(),
+      format: format((obj: TransformableInfo) => {
+        if (!redactionPattern || !obj) {
+          return obj;
+        }
+
+        obj[MESSAGE] = obj[MESSAGE]?.replace?.(redactionPattern, '[REDACTED]');
+        obj.message = obj.message.replace?.(redactionPattern, '[REDACTED]');
```
Comment (benjdlambert) [inline packages/backend-app-api/src/logging/WinstonLogger.ts:96]: I just did this for backwards compatibility if anyone has written anything that doesn't use the `MESSAGE` symbol, but maybe we could actually remove it @Rugvip?

Comment (Rugvip) [reply packages/backend-app-api/src/logging/WinstonLogger.ts:96]: Yeah I think it should be safe to skip this tbh

File: plugins/scaffolder-backend/src/scaffolder/tasks/logger.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ import { JsonObject } from '@backstage/types';
 import { Format, TransformableInfo } from 'logform';
 import Transport, { TransportStreamOptions } from 'winston-transport';
 import { Logger, format, createLogger, transports } from 'winston';
+import { MESSAGE } from 'triple-beam';
```
Comment (Rugvip) [inline plugins/scaffolder-backend/src/scaffolder/tasks/logger.ts:24]: Wanna note that this is implemented with `Symbol.for(...)`, so duplication is not an issue.

**Review by Rugvip — COMMENTED** [review]
🎉

**Pull request merged**